### DISCUSSION TITLE
shortcode: show code snippets from file

### DISCUSSION THREAD

**Discussion by tied** [McShelby/hugo-theme-relearn#179] [Q&A] [answered]
The [Shadocs](https://shadocs.netlify.app/shortcodes/snippet/) Hugo theme has bundled a snippets shortcut. It would be awesome if something similar could be implemented into the relearn theme.

**Use Case**
I use the Hugo site as documentation for my Ansible Roles / Playbooks and things related to that.
One part of that are the default variables of each role. At the moment the Hugo page has an exact copy of the vars file. If things are changed in the Role variables these changes are not reflected in the Hugo site.

My idea was to add my Ansible roles as git sub modules (similar to the theme files itself) and then include the file directly with the snippet shortcode. And also benefit from the current code highlighting.

Bonus: A snippet can be used in Tabs :) 

I've simply tried to copy the shortode from Shadocs theme and technically it would work but it has some trade-offs like the tabs that are not working, etc. But the Include itself would work.

Thanks :)

**Comment by McShelby** [comment, marked as answer]
At the moment I don't plan to introduce such a feature. But you may work around with the existing shortcodes [`include`](https://mcshelby.github.io/hugo-theme-relearn/shortcodes/include/) and [`tabs`](https://mcshelby.github.io/hugo-theme-relearn/shortcodes/tabs/) or a combination of both like:

Plain file include (will be interpreted as markdown) [as used in the exampleSite](https://mcshelby.github.io/hugo-theme-relearn/basics/history/):
````
{{% include "basics/CHANGELOG.md" %}}
````

File include as code:
````
```json
{{% include "example.json" %}}
```
````

Plain file include in a tab:
````
{{< tabs >}}
{{% tab name="The changelog" %}}
{{% include "basics/CHANGELOG.md" %}}
{{% /tab %}}
{{< /tabs >}}
````

File include as code in a tab:
````
{{< tabs >}}
{{% tab name="json" %}}
```json
{{% include "example.json" %}}
```
{{% /tab %}}
{{< /tabs >}}
````

Let me know, if this solves your case.
